code/modules/mob/living/silicon/robot/drone/drone.dm

```diff
@@ -154,15 +166,17 @@
 			return
 
 		else
-			user.visible_message("<span class='warning'>\the [user] swipes [user.p_their()] ID card through [src], attempting to shut it down.</span>", "<span class='warning'>You swipe your ID card through \the [src], attempting to shut it down.</span>")
+			var/confirm = alert("Using your ID on a Maintenance Drone will shut it down, are you sure you want to do this?", "Disable Drone", "Yes", "No")
```

Comment:
This isnt going to pass until this is changed to have a range check, or a 3 seconds timeout.

Reply:
I'll try to figure that out tomorrow, too late right now.

Reply:
Pretty sure that does it.